source/trigger/position.go

```diff
@@ -0,0 +1,69 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package trigger
+
+import (
+	"encoding/json"
+	"fmt"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+// Mode defines an iterator mode.
+type Mode string
+
+const (
+	ModeSnapshot = "snapshot"
+	ModeCDC      = "cdc"
+)
+
+// Position is a position.
+type Position struct {
+	// Mode is the current iterator mode.
+	Mode Mode `json:"mode"`
+	// LastProcessedVal is the most recently processed value of an orderingColumn column.
+	LastProcessedVal any `json:"lastProcessedVal"`
+	// CreatedAt is the time of the first connector start (in the format hhmmss).
+	CreatedAt string `json:"createdAt"`
+}
+
+// ParseSDKPosition parses sdk.Position and returns Position.
+func ParseSDKPosition(position sdk.Position) (*Position, error) {
+	var pos Position
+
+	if position == nil {
+		return &Position{
+			Mode:      ModeSnapshot,
+			CreatedAt: time.Now().Format("150405"),
+		}, nil
+	}
+
+	if err := json.Unmarshal(position, &pos); err != nil {
+		return nil, fmt.Errorf("unmarshal sdk.Position into Position: %w", err)
+	}
+
+	return &pos, nil
```

Comment:
What do you think about this?
```suggestion
    pos := new(Position)
	if err := json.Unmarshal(position, pos); err != nil {
		return nil, fmt.Errorf("unmarshal sdk.Position into Position: %w", err)
	}

	return pos, nil
```
Google's style guide has a nice example of such a moment at the end of the ["Non-pointer zero values" section](https://google.github.io/styleguide/go/best-practices#non-pointer-zero-values).

Reply:
Good point 👍 
I should probably read it 🤔